File: Quick-Sorting.js
class Solution {
    quickSort(nums) {
        if (nums.length <= 1) {
            return nums;
        }

        let pivot = nums[0];               // correct pivot
        let lesser = nums.slice(1).filter(x => x <= pivot);
        let higher = nums.slice(1).filter(x => x > pivot);

        return [
            ...this.quickSort(lesser),
            pivot,
            ...this.quickSort(higher)
        ];
    }
}

const n = [7, 4, 1, 5, 3];
console.log(new Solution().quickSort(n));

File: Quick-Sorting.py
class Solution:
    def quickSort(self, nums):
        if len(nums)<=1:
            return nums 
        else:
            i= nums[0]
            lesser= [x for x in nums[1:] if x<= i]
            higher= [x for x in nums[1:] if x> i]
            return self.quickSort(lesser) + [i] + self.quickSort(higher) 
n=input().strip("[]")
n=list(map(int, n.split(",")))
print(Solution().quickSort(n))
